GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -110,18 +118,50 @@ public bool TryGetLooseObjectId(string directoryName, string filePath, out strin
         /// Creates a pack file from loose objects
         /// </summary>
         /// <returns>The number of loose objects added to the pack file</returns>
-        public int CreateLooseObjectsPackFile()
+        public bool TryCreateLooseObjectsPackFile(out int objectsAddedToPack)
         {
-            int objectsAddedToPack = 0;
+            int localObjectCount = 0;
 
             GitProcess.Result result = this.RunGitCommand(
                 (process) => process.PackObjects(
                     "from-loose",
                     this.Context.Enlistment.GitObjectsRoot,
-                    (StreamWriter writer) => objectsAddedToPack = this.WriteLooseObjectIds(writer)),
+                    (StreamWriter writer) => localObjectCount = this.WriteLooseObjectIds(writer)),
                 nameof(GitProcess.PackObjects));
 
-            return objectsAddedToPack;
+            if (result.ExitCodeIsSuccess)
+            {
+                objectsAddedToPack = localObjectCount;
+                return true;
+            }
+            else
+            {
+                objectsAddedToPack = 0;
+                return false;
+            }
+        }
+
+        public int ClearCorruptLooseObjects()
+        {
+            int numBadObjects = 0;
+            foreach (string objectId in this.LooseObjectsBatch(this.MaxLooseObjectsInPack))
+            {
+                if (!this.Context.Repository.TryGetIsBlob(objectId, out bool isBlob) ||

```

Comment:
Can you confirm how we get into this state?
1. The loose object is missing? (TryGetIsBlob fails)
2. The object is corrupt? (fails the !isBlob)

Some comments here would be useful

Reply:
The way we get into this state is unclear, but the issue is that the file exists and is corrupt, so Git can't do anything about it.

Most likely, these are corruptions from many versions ago before we wrote loose objects safely. (See `GitObjects.WriteLooseObject()` for the details on our current writes.) But it could also be from disk problems.

Reply:
Makes sense, perhaps `isValidBlob` rather than just `isBlob `may be helpful.  
A one line comment would also do the trick.